packages/playwright-test/src/runner/runner.ts

```diff
@@ -87,12 +87,13 @@ export class Runner {
     }
 
     const taskStatus = await taskRunner.run(testRun, deadline);
+    const maxFailuresReached = testRun.phases.some(p => p.dispatcher.hasReachedMaxFailures());
```

Comment:
Now that I see this line, there is a bug where we can have up to 99 failures in each dispatcher 😄  Let's fix it as well.

Reply:
Filed as https://github.com/microsoft/playwright/issues/26344.